sagan/models/sagan.py

```diff
@@ -0,0 +1,75 @@
+import tensorflow as tf
+from tensorflow import keras
+
+from models.generator import build_generator
+from models.discriminator import build_discriminator
+from sagan.utils import w_loss
+
+class SelfAttentionGAN(keras.Model):
+
+    def __init__(self, latent_dim, gp_weight, **kwargs):
+        super(SelfAttentionGAN, self).__init__(**kwargs)
+        self.latent_dim = latent_dim
+        self.gp_weight = gp_weight
+        self.generator = build_generator(IMAGE_SIZE, latent_dim)
```

Comment:
As mentioned in https://github.com/soumik12345/self-attention-gan/pull/3#discussion_r744122023, the config object should be passed to the constructor instead of using constants. Refer to https://github.com/soumik12345/point-cloud-segmentation